src/main/fc/fc_rc.c

```diff
@@ -295,15 +306,17 @@ FAST_CODE uint8_t processRcSmoothingFilter(void)
                     derivativeCutoffFrequency = (derivativeCutoffFrequency == 0) ? defaultDerivativeCutoffFrequency : derivativeCutoffFrequency;
 
                     const float dT = targetPidLooptime * 1e-6f;
-                    for (int i = 0; i < interpolationChannels; i++) {
-                        switch (rxConfig()->rc_smoothing_input_type) {
-                            case RC_SMOOTHING_INPUT_BIQUAD:
-                                biquadFilterInitLPF(&rcCommandFilterBiquad[i], filterCutoffFrequency, targetPidLooptime);
-                                break;
-                            case RC_SMOOTHING_INPUT_PT1:
-                            default:
-                                pt1FilterInit(&rcCommandFilterPt1[i], pt1FilterGain(filterCutoffFrequency, dT));
-                                break;
+                    for (int i = 0; i < PRIMARY_CHANNEL_COUNT; i++) {
+                        if ((1 << i) & interpolationChannels) {
+                            switch (rxConfig()->rc_smoothing_input_type) {
+                                case RC_SMOOTHING_INPUT_BIQUAD:
+                                    biquadFilterInitLPF(&rcCommandFilterBiquad[i], filterCutoffFrequency, targetPidLooptime);
+                                    break;
+                                case RC_SMOOTHING_INPUT_PT1:
+                                default:
+                                    pt1FilterInit(&rcCommandFilterPt1[i], pt1FilterGain(filterCutoffFrequency, dT));
+                                    break;
+                            }
                         }
```

Comment:
Sorry I didn't catch this on the first go round.  I stopped testing after I saw the channel problem.

The filter-based portion in `processRcSmoothingFilter()` is not working correctly.  Any channels that are not selected for smoothing (still uses `rc_interp_ch`) are getting 0 values for the channel.  The problem is by adding the check to:

`if (filterInitialized && (1 << updatedChannel) & interpolationChannels) {`

it's falling into the else clause that's only meant to be used when the filters are not initialized.  Unfortunately `lastRxData[]` is not updated due to logic added to that section (which is okay):
```
   if (isRXDataNew) {
        for (int i = 0; i < PRIMARY_CHANNEL_COUNT; i++) {
            if ((1 << i) & interpolationChannels) {
                lastRxData[i] = rcCommand[i];
            }
        }
```

What we really want to do is simply not modify `rcCommand[updatedChannel]` at all if the channel is not smoothed.  rcCommand already has the new value from the rx so we leave it unmodified.  I would simply wrap the entire content with the new condition to skip it if the channel isn't smoothed.  I tested with this version and it works as expected:
```
    for (updatedChannel = 0; updatedChannel < PRIMARY_CHANNEL_COUNT; updatedChannel++) {
        if ((1 << updatedChannel) & interpolationChannels) {
            if (filterInitialized) {
                switch (rxConfig()->rc_smoothing_input_type) {
                    case RC_SMOOTHING_INPUT_BIQUAD:
                        rcCommand[updatedChannel] = biquadFilterApply(&rcCommandFilterBiquad[updatedChannel], lastRxData[updatedChannel]);
                        break;
                    case RC_SMOOTHING_INPUT_PT1:
                    default:
                        rcCommand[updatedChannel] = pt1FilterApply(&rcCommandFilterPt1[updatedChannel], lastRxData[updatedChannel]);
                        break;
                }
            } else {
                // If filter isn't initialized yet then use the actual unsmoothed rx channel data
                rcCommand[updatedChannel] = lastRxData[updatedChannel];
            }
        }
    }
```

Reply:
't was too good to be true...